source/common/upstream/upstream_impl.h

```diff
@@ -72,6 +72,42 @@ class HealthCheckHostMonitorNullImpl : public HealthCheckHostMonitor {
   void setUnhealthy(UnhealthyType) override {}
 };
 
+/**
+ * Implementation of LoadMetricStats.
+ *
+ * TODO(pianiststickman): this implementation puts a copy of the stat name into every host that
+ * receives a copy of that metric. This can be improved by putting a single copy of the stat name
```

Comment:
Also note here that it takes a lock on the hot-path.

I really think this should not be deployed as is, but I understand why you want to add the interfaced and iterate on the perf.

Reply:
Done.